### PR TITLE
Add governance to update the referral program

### LINKED ISSUE #9012: Implement referral programs governance

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #9012

### CODE REVIEW

File: protos/sources/vega/governance.proto [hunk 1]
```diff
@@ -546,6 +550,11 @@ message Vote {
   string total_equity_like_share_weight = 7;
 }
 
+message UpdateReferralProgram {
+  // Configuration for change of the referral program.
```
Comment (candida-d) [inline protos/sources/vega/governance.proto:554]:
```suggestion
  // Configuration for change to update a referral program.
```

File: protos/sources/vega/vega.proto [hunk 2]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
 }
+
+message ReferralProgram {
+  // Defined tiers in increasing order. First element will give Tier 1, second
+  // element will give Tier 2, and so on.
+  repeated BenefitTier benefit_tiers = 1;
+  //  Timestamp as Unix time in seconds after which when the current epoch ends, the
+  // programs status will become STATE_CLOSED and benefits will be disabled.
+  int64 end_of_program_timestamp = 2;
+  // The number of epochs over which to evaluate a referral sets running volume.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1582]:
```suggestion
  // The number of epochs over which to evaluate a referral set's running volume.
```

File: protos/sources/vega/vega.proto [hunk 3]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
+}
+
+message BenefitTier {
+  // The required running taker volume in quantum units for parties to access this
+  // tier.
+  string minimum_running_notional_taker_volume = 1;
+  // The required number of epochs a party must have been in a referral set to
+  // access this tier.
+  string minimum_epochs = 2;
+  // The proportion of the referees taker fees to be rewarded to the referrer
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1593]:
```suggestion
  // The proportion of the referee's taker fees to be rewarded to the referrer
```

File: protos/sources/vega/vega.proto [hunk 4]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
+  string referral_reward_factor = 3;
+  // The proportion of the referees taker fees to be discounted
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1595]:
```suggestion
  // The proportion of the referee's taker fees to be discounted
```

File: protos/sources/vega/vega.proto [hunk 5]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
+  //  Timestamp as Unix time in seconds after which when the current epoch ends, the
+  // programs status will become STATE_CLOSED and benefits will be disabled.
+  int64 end_of_program_timestamp = 2;
+  // The number of epochs over which to evaluate a referral sets running volume.
+  uint64 window_length = 3;
+}
+
+message BenefitTier {
+  // The required running taker volume in quantum units for parties to access this
+  // tier.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1588]:
```suggestion
  // The required running notional taker volume in quantum units for parties 
  // to access this tier.
```

**Review by candida-d — COMMENTED** [review]
a few more changes to the doc strings

File: protos/sources/vega/vega.proto [hunk 6]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
 }
+
+message ReferralProgram {
+  // Defined tiers in increasing order. First element will give Tier 1, second
+  // element will give Tier 2, and so on.
+  repeated BenefitTier benefit_tiers = 1;
+  //  Timestamp as Unix time in seconds after which when the current epoch ends, the
+  // programs status will become STATE_CLOSED and benefits will be disabled.
+  int64 end_of_program_timestamp = 2;
+  // The number of epochs over which to evaluate a referral set's running volume.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1582]:
```suggestion
  // Number of epochs over which to evaluate a referral set's running volume.
```

Removing "The" based on API docs style guide

File: protos/sources/vega/vega.proto [hunk 7]
```diff
@@ -1571,3 +1571,27 @@ message VectorValue {
...
+message BenefitTier {
+  // The required running notional taker volume in quantum units for parties 
+  // to access this tier.
+  string minimum_running_notional_taker_volume = 1;
+  // The required number of epochs a party must have been in a referral set to
+  // access this tier.
+  string minimum_epochs = 2;
+  // The proportion of the referee's taker fees to be rewarded to the referrer
+  string referral_reward_factor = 3;
+  // The proportion of the referee's taker fees to be discounted
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1595]:
```suggestion
  // Required running notional taker volume in quantum units for parties 
  // to access this tier.
  string minimum_running_notional_taker_volume = 1;
  // Required number of epochs a party must have been in a referral set to
  // access this tier.
  string minimum_epochs = 2;
  // Proportion of the referee's taker fees to be rewarded to the referrer
  string referral_reward_factor = 3;
  // Proportion of the referee's taker fees to be discounted
```

**Review by candida-d — COMMENTED** [review]
I missed some things in the original review

**Review by peterbarrow — APPROVED** [review]

**Review by gordsport — APPROVED** [review]

**Review by cdummett — APPROVED** [review]

**Pull request merged**